File: src/frc/robot/Robot.java
package frc.robot;

import frc.robot.imp.TimedRobot;
import edu.wpi.first.wpilibj2.command.Command;
import edu.wpi.first.wpilibj2.command.CommandScheduler;
import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;

public class Robot extends TimedRobot {
  private ShuffleboardTab tab;
  private RobotContainer robotContainer;
  private Command autoCommand;
  public Robot() {
    tab = Shuffleboard.getTab("Master");
    robotContainer = new RobotContainer();
    autoCommand = null;
  }

  public void robotInit() {
  }

  public void robotPeriodic() {
    CommandScheduler.getInstance().run();
  }

  public void disabledInit() {
    robotContainer.swerve.stop();
    robotContainer.useVision(true);
    robotContainer.vision.on();
    robotContainer.lights.off();
  }

  public void disabledPeriodic() {
    robotContainer.loadAuto();
  }

  public void autonomousInit() {
    robotContainer.vision.off();
    autoCommand = robotContainer.currentAuto;
    autoCommand.schedule();
  }

  public void autonomousPeriodic() {}

  public void teleopInit() {
    robotContainer.vision.off();
    robotContainer.useVision(true);
    robotContainer.lights.off(robotContainer.swerve);
    if(autoCommand != null) {
      autoCommand.cancel();
    }
  }

  public void teleopPeriodic() {}

  public void testInit() {
    CommandScheduler.getInstance().cancelAll();
  }

  public void testPeriodic() {}
}

File: src/main/kotlin/frc/robot/Robot.kt
package frc.robot;

import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj2.command.Command;
import edu.wpi.first.wpilibj2.command.CommandScheduler;
import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard
import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab
import edu.wpi.first.networktables.NetworkTableInstance

val tab: ShuffleboardTab = Shuffleboard.getTab("Master")

class Robot : TimedRobot() {
  private val robotContainer = RobotContainer(tab);
  private var autoCommand = robotContainer.getAutonomousCommand();

  override fun robotInit() {
  }

  override fun robotPeriodic() {
    CommandScheduler.getInstance().run();
  }

  override fun disabledInit() {
  }

  override fun disabledPeriodic() {}

  override fun autonomousInit() {
    autoCommand = robotContainer.getAutonomousCommand();
    autoCommand.schedule();
  }

  override fun autonomousPeriodic() {}

  override fun teleopInit() {
    autoCommand.cancel();
  }

  override fun teleopPeriodic() {}

  override fun testInit() {
    CommandScheduler.getInstance().cancelAll();
  }

  override fun testPeriodic() {}
}
